src/server.c

```diff
@@ -130,22 +137,51 @@ void serverLogRaw(int level, const char *msg) {
         int off;
         struct timeval tv;
         int role_char;
+        const char * role;
         pid_t pid = getpid();
 
         gettimeofday(&tv,NULL);
         struct tm tm;
         nolocks_localtime(&tm,tv.tv_sec,server.timezone,server.daylight_active);
-        off = strftime(buf,sizeof(buf),"%d %b %Y %H:%M:%S.",&tm);
-        snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
+        const char* timestamp_format = LOG_TIMESTAMP_FORMATS[server.log_timestamp_format];
+        if (server.log_timestamp_format == LOG_TIMESTAMP_UNIX) {
+            // UNIX timestamp
+            snprintf(buf,sizeof(buf),timestamp_format,tv.tv_sec);
+        } else {
+            // Other timestamps
+            off = strftime(buf,sizeof(buf),timestamp_format,&tm);
+            // Add millisecond
+            snprintf(buf+off,sizeof(buf)-off,"%03d",(int)tv.tv_usec/1000);
+            // Add time zone
+            if (server.log_timestamp_format == LOG_TIMESTAMP_ISO8601){
+                char tzbuf[6];
+                strftime(tzbuf, sizeof(tzbuf), "%z", &tm);
+                // strftime(tzbuf, sizeof(tzbuf), "%z", &tm);
+                strncat(buf, tzbuf, sizeof(buf) - strlen(buf) - 1);
+            }
+        }
```

Comment:
There is no point of having the unix timestamp format in the `LOG_TIMESTAMP_FORMATS` array when we also have an `if` for the unix format. We can just use `"%ld"` directly here.

For the other formats, I don't like the `LOG_TIMESTAMP_FORMATS` array either, because it has an implicit dependency by index to the config enum. I think it's better to just use switch and if/else-if on `server.log_timestamp_format` here.